### PR TITLE
README: update wording to feature cloud more heavily

### PR DESCRIPTION
**Pull request by benesch**
We've had several reports of users getting confused by our README and come away thinking that Materialize is primarily a single node system with no fault tolerance or horizontal scalability.

This commit is a small reconfiguration of our README to minimize that confusion.

It also bundles some other small improvements:

  * More links to contact us and join our Slack
  * A fix to the contact us link in the LICENSE file
  * Clarification that `HAVING` is not an aggregate function
  * Use of the correct "good for external contributors" label
  * Less dire warnings about using ORMs, since we've gotten increasingly good at supporting ORMs over the years

### Motivation

   * This PR refactors existing documentation.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - n/a

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,25 +4,33 @@
 
 [<img src="https://github.com/MaterializeInc/materialize/assets/23521087/39270ecb-7ac4-4829-b98b-c5b5699a16b8" width=35%>](https://materialize.com)
 
-Materialize is a **data warehouse** purpose-built for **operational workloads**
-where an analytical data warehouse would be too slow, and a stream processor
-would be too complicated.
+Materialize is a cloud-native **data warehouse** purpose-built for **operational
```
Comment (sjwiesman) [inline README.md:7]: Should there be a reference to persistence storage at S3 scale?

File: README.md [hunk 2]
```diff
@@ -4,25 +4,33 @@
...
+  * [Join our Slack](https://materialize.com/s/chat)
+  * [Send us mail](https://materialize.com/contact/)
+
 ## About
 
 Materialize is designed to help you interactively explore your streaming data, perform analytics against live relational data, or increase data freshness while reducing the load of your dashboard and monitoring tasks. The moment you need a refreshed answer, you can get it in milliseconds.
 
 It focuses on providing correct and [consistent](https://materialize.com/docs/overview/isolation-level/) answers with minimal latency, and does not ask you to accept either approximate answers or eventual consistency. Whenever Materialize answers a query, that answer is the correct result on some specific (and recent) version of your data. Materialize does all of this by recasting your SQL queries as *dataflows*, which can react efficiently to changes in your data as they happen.
 
+Our fully managed service is cloud native, featuring **high availability**, via
```
Comment (sjwiesman) [inline README.md:30]: Should there be a reference to persistent storage at S3 scale?

Comment (benesch) [reply README.md:30]: That's a great idea. Updated.

File: README.md [hunk 3]
```diff
@@ -4,25 +4,34 @@
...
```
Comment (morsapaes) [inline README.md:30]:
```suggestion
Our fully managed service is cloud-native, featuring **high availability**, via
```

File: README.md [hunk 4]
```diff
@@ -4,25 +4,34 @@
...
+multi-active replication, **horizontal scalability**, by seamlessly scaling
+dataflows across multiple machines, and **near infinite storage**, by
```
Comment (morsapaes) [inline README.md:32]:
```suggestion
compute across multiple machines, and **near infinite storage**, by
```

File: README.md [hunk 5]
```diff
@@ -35,11 +44,12 @@ Once you've got the data in, define views and perform reads via the PostgreSQL p
...
+-   Joins, joins, joins! Materialize supports multi-column join conditions, multi-way joins, self-joins, cross-joins, inner joins, outer joins, etc.
 -   Delta-joins avoid intermediate state blowup compared to systems that can only plan nested binary joins - tested on joins of up to 64 relations.
 -   Support for subqueries. Materialize's SQL optimizer performs subquery decorrelation out-of-the-box, avoiding the need to manually rewrite subqueries into joins.
--   Materialize supports streams that contain CDC data (currently supporting the [Debezium](https://debezium.io/blog/2017/09/25/streaming-to-another-database/) format). Materialize can incrementally maintain views in the presence of arbitrary inserts, updates, and deletes. No asterisks.
--   All the aggregations. `GROUP BY` , `MIN`, `MAX`, `COUNT`, `SUM`, `STDDEV`, `HAVING`, etc.
+-   Materialize supports streams that contain CDC data (e.g., [Debezium](https://debezium.io/blog/2017/09/25/streaming-to-another-database/), [Striim](https://striim.com)). Materialize can incrementally maintain views in the presence of arbitrary inserts, updates, and deletes. No asterisks.
```
Comment (morsapaes) [inline README.md:50]:
```suggestion
-   Materialize supports streams that contain CDC data (e.g., [Debezium](https://debezium.io/blog/2017/09/25/streaming-to-another-database/), [direct PostgreSQL CDC](https://materialize.com/docs/sql/create-source/postgres/)). Materialize can incrementally maintain views in the presence of arbitrary inserts, updates, and deletes. No asterisks.
```

File: README.md [hunk 6]
```diff
@@ -103,17 +113,24 @@ Stream inserts, updates, and deletes on the underlying tables (`lineitem` and `s
...
 
 Check out [our documentation](https://materialize.com/docs/).
 
 ## License
 
-Materialize is source-available and [licensed](LICENSE) under the BSL 1.1, converting to the open-source Apache 2.0 license after 4 years. As stated in the BSL, Materialize is free forever on a single node.
+Materialize is provided primarily as a [fully managed cloud
+service](https://materialize.com/pricing/), with proprietary cloud-native
+features like horizontal scalability, high availability, and a web management
+console included in the cost.
```
Comment (morsapaes) [inline README.md:125]:
```suggestion
Materialize is provided primarily as a fully managed cloud service with
[credit-based pricing](https://materialize.com/pricing/). It offers proprietary
cloud-native features like horizontal scalability, high availability, and a web
console included in the cost.
```

File: README.md [hunk 7]
```diff
@@ -103,17 +113,24 @@ Stream inserts, updates, and deletes on the underlying tables (`lineitem` and `s
...
 
-Materialize is also available as [a paid cloud service](https://materialize.com/pricing/) with additional features such as high availability via multi-active replication.
+However, we're big believers in advancing the frontier of human knowledge. To
+that end, the source code of the standalone database engine is publicly
+available, in this repository, and [licensed](LICENSE) under the BSL 1.1,
+converting to the open-source Apache 2.0 license after 4 years. As stated in the
+BSL, use of the standalone database engine on a single node is free forever.
+Please be warned that this deployment model is *not* suitable for production use
+and we cannot offer support for it.
```
Comment (morsapaes) [inline README.md:133]:
```suggestion
However, we're big believers in advancing the state-of-the-art of database and
streaming technologies. To that end, the source code of the standalone database
engine is publicly available, in this repository, and [licensed](LICENSE) under
the BSL 1.1, converting to the open-source Apache 2.0 license after 4 years. As
stated in the BSL, use of the standalone database engine on a single node is
free forever. Please be warned that this deployment model is *not* suitable for
production use and we cannot offer support for it.
```

**Review by morsapaes — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (benesch) [reply README.md:30]: The noun here ("fully managed service") comes before the compound adjective ("cloud native"), so typical style is to not use a hyphen!

File: README.md (hunk 5 above)
Comment (benesch) [reply README.md]: Maybe we just strike the first sentence completely? The second sentence is the real point ("Materialize can incrementally maintain views in the presence of arbitrary inserts, updates, and deletes.") and is punchier without the first sentence. The paragraph above the list also already mentions the ways you can get data into Materialize these days (Kafka, PostgreSQL, webhooks).

File: README.md (hunk 4 above)
Comment (morsapaes) [reply README.md:32]: Oh, I didn't re-read the whole thing. We could remove references to dataflows and more generically use "compute", which doesn't presuppose knowledge of Materialize internals; but not something I feel strongly about.

File: README.md (hunk 7 above)
Comment (morsapaes) [reply README.md:133]: :heavybreathing:

**Review by ay-materialize — APPROVED** [review]
lgtm

File: README.md (hunk 4 above)
Comment (benesch) [reply README.md:32]: Yeah, might be worth doing, but also I don't mind that the GitHub README goes a little deeper into the technical weeds than other things. Just gonna leave as is for now and let someone else address in the future.

**Pull request merged**